src/main/scala/com/imaginea/activegrid/core/discovery/models/Filter.scala

```diff
@@ -0,0 +1,53 @@
+package com.imaginea.activegrid.core.discovery.models
+
+import com.imaginea.activegrid.core.models.BaseEntity
+
+/**
+ * Created by ranjithrajd on 12/10/16.
+ */
+case class Filter(override val id: Option[Long]
+                  , filterType: FilterType
+                  , values: List[String]
+                  , condition: Condition) extends BaseEntity
+
+sealed trait FilterType {
+  def name: String
+  override def toString: String = name
+}
+
+case object Tags extends FilterType { val name = "TAGS" }
+case object KeypairNames extends FilterType { val name = "KEYPAIR_NAMES" }
+case object SecurityGroups extends FilterType { val name = "SECURITY_GROUPS" }
+case object IpRanges extends FilterType { val name = "IP_RANGES" }
+case object InstanceIds extends FilterType { val name = "INSTANCE_IDS" }
+case object Status extends FilterType { val name = "STATUS" }
+
+object FilterType {
+  implicit def toFilterType(name: String): FilterType = name match {
+    case "TAGS" => Tags
+    case "KEYPAIR_NAMES" => KeypairNames
+    case "SECURITY_GROUPS" => SecurityGroups
+    case "IP_RANGES" => IpRanges
+    case "INSTANCE_IDS" => InstanceIds
+    case "STATUS" => Status
+  }
+}
+
+sealed trait Condition {
+  def name: String
+  override def toString: String = name
```

Comment:
Condition is part of Filter case class. It was written as Enum in java and in scala we have moved it to the sealed Trait.

When we are persisting this Condition type that should be converted as String.
So I am holding name property and overriding it in the all other sub types.
I did refer this tutorial below link,
http://stackoverflow.com/questions/1898932/case-objects-vs-enumerations-in-scala